.github/workflows/test.yaml

```diff
@@ -19,34 +19,32 @@ jobs:
     timeout-minutes: 20
     strategy:
       matrix:
-        python_version:
-        - '3.9'
-        subdir:
-        - bert
-        - cifar
-        - deeplab
-        - llm
-        - resnet
+        include:
+        - name: bert-pytest
+          container: mosaicml/pytorch:1.13.1_cpu-python3.10-ubuntu20.04
+          subdir: bert
+        - name: cifar-pytest
+          container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
+          subdir: cifar
+        - name: deeplab-pytest
+          container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
+          subdir: deeplab
+        - name: llm-pytest
+          container: mosaicml/pytorch:1.13.1_cpu-python3.10-ubuntu20.04
+          subdir: llm
+        - name: resnet-pytest
+          container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
+          subdir: resnet
+    name: ${{ matrix.name }}
+    container: ${{ matrix.container }}
     steps:
     - uses: actions/checkout@v3
-    - uses: actions/setup-python@v4
-      with:
-        python-version: ${{ matrix.python_version }}
-    - name: Cache pip
-      uses: actions/cache@v3
-      with:
-        # This path is specific to Ubuntu
-        path: ~/.cache/pip
-        # Look to see if there is a cache hit for the corresponding requirements file
-        key: ${{ runner.os }}-pip-${{ hashFiles('setup.py', 'requirements.txt',  '${{
-          matrix.subdir }}/requirements.txt') }}
-        restore-keys: |
-          ${{ runner.os }}-pip-
-          ${{ runner.os }}-
     - name: Setup
       run: |
         set -ex
-        python -m pip install --upgrade pip wheel
+        export PATH=/composer-python:$PATH
+        python -m pip install --upgrade 'pip<23' wheel
     - name: Test
       run: |
+        export PATH=/composer-python:$PATH
```

Comment:
is there a clean way to avoid adding this to the path? I doubt users will modify their PATH, so I'm worried about the CI not reflecting what users experience if they try to install and use the repo

Reply:
So the path adding is only needed because composer images do weird stuff with python. I think we're going to fix this soon and then this command goes away. That being said, the actual lint/test is still running inside a venv, so that should be basically what a user would experience.